lua/neo-tree/defaults.lua

```diff
@@ -491,6 +493,61 @@ local config = {
       },
     },
   },
+  document_symbols = {
+    renderers = {
+      root = {
+        {"indent"},
+        {"icon", default="C" },
+        {"name", zindex = 10},
+      },
+      symbol = {
+        {"indent"},
+        {"icon", default="C" },
+        {"container",
+        content = {
+          {"name", zindex = 10},
+          {"kind_name", zindex = 20, align = "right"},
+          {"kind_icon", zindex = 20, align = "right"},
+          }
+        }
+      },
+    },
+    window = {
+      mappings = {
+        ["<cr>"] = "toggle_node",
+        ["o"] = "jump_to_symbol",
+      },
+    },
+    server_filter = "first",
+    kinds = {
+      [1] = { name = "File", icon = "", hl = "Tag" },
```

Comment:
Why are kinds keyed by numbers? It seems more natural that what is currently the `name` field would be the key. Do the numbers have a special meaning?

Reply:
According to the [LSP specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol), the kinds are defined as enums and are returned as integers in the server. 

The conventional list of kinds looks like this
<details>
<summary>Kinds list</summary>
```
export namespace [SymbolKind](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind) {
	export const File = 1;
	export const Module = 2;
	export const Namespace = 3;
	export const Package = 4;
	export const Class = 5;
	export const Method = 6;
	export const Property = 7;
	export const Field = 8;
	export const Constructor = 9;
	export const Enum = 10;
	export const Interface = 11;
	export const Function = 12;
	export const Variable = 13;
	export const Constant = 14;
	export const String = 15;
	export const Number = 16;
	export const Boolean = 17;
	export const Array = 18;
	export const Object = 19;
	export const Key = 20;
	export const Null = 21;
	export const EnumMember = 22;
	export const Struct = 23;
	export const Event = 24;
	export const Operator = 25;
	export const TypeParameter = 26;
}
```
</details>

It seems that other servers could define more kinds than this, for example in [here](https://github.com/glepnir/lspsaga.nvim/blob/fb476086012e18e0001c3dcc5b18fd34a847e5fe/lua/lspsaga/lspkind.lua#L59), so I make the kind list configurable by index to accommodate these additional kinds.

Reply:
I think it would be a good idea to define the common kinds keyed by `name` as you said and have an option for additional symbols.